### PR TITLE
Updated ProfileUpdate image upload request with useMutation

### PR DESCRIPTION
**Pull request by filoscoder**
## Specify project
Related on both side: **Cliente**, **Server**

## Description
- Updated `singleUpload` mutation & `azure blob`
- Created mutation & `useMutation` instead of `fetch`
- Added fetchGraphQL logic to handle `uploadables`
- Fixed uploading `file` with `Blob` constructor

|**web**|**ios**|
|---|---|
|web_singleupload|ios_singleupload
|  
## Related Issues
**Issues on Web**
1) The filename extracted from the uploaded file `uri` is **weird** (doesn't look like a local filename), so I deduce this will lead to being unable to set the `File Blob` object name (`file.name = fileName` this is not working. When the file argument is printed it can found the name is `blob`, the default value of `filename` property)
- **CORRECT**
스크린샷 2020-08-28 18 53 31

- **INCORRECT**
스크린샷 2020-08-28 18 56 49

2) When `pressProfileImage` is fired there is 3 options showing: a) 'TAKE_A_PICTURE', b) 'SELECT_FROM_ALBUM', c) 'CANCEL'
If the user is following to search and pick an image from the `library` nothing happens. This is because when this option is pressed the `status` of `Permissions.askAsync(Permissions.CAMERA_ROLL)` returns `'undetermined'`
web_issue2

**Android issue**
- Unable to test image upload feature on Android emulator due to unexpected "Network error" (working with local server at `http://localhost:4000/graphql`)

## Tests

I added the following tests:

_Replace this with a list of the tests that you added as part of this PR. A change in behavior with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage](https://codecov.io/gh/dooboolab/hackatalk)._

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn lint && yarn tsc`
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by filoscoder** [comment]
Please take a look at the changed files @hyochan!
If you have any suggestions about the issues, please let me know.

**Comment by hyochan** [comment]
Please check the failing `CI`!

File: server/src/utils/azure.ts [hunk 1]
```diff
@@ -16,19 +16,21 @@ export const uploadFileToAzureBlobFromStream = (
...
+               `${destDir}${destFile}`,
+               function(error, resultUpload) {
+                 if (!error) {
+                   resolve(resultUpload);
+                   return;
+                 }
+                 reject(error);
+               },
+      ),
     );
```
Comment (hyochan) [inline server/src/utils/azure.ts:34]: Looks like the indentations are not correct!

File: client/src/components/screen/ProfileUpdate.tsx [hunk 2]
```diff
@@ -121,42 +137,48 @@ const Screen: FC<Props> = () => {
...
     const fileType = fileTypeMatch ? `image/${fileTypeMatch[1]}` : 'image';
-    const data: FormData = new FormData();
-    const token = await AsyncStorage.getItem('token');
-    data.append('profile', {
-      uri: uri,
+    const file = new Blob([uri], {
       type: fileType,
-      name: fileName,
+      endings: 'native',
     });
```
Comment (hyochan) [inline client/src/components/screen/ProfileUpdate.tsx:147]: Wow so you found the way to do this directly without `FormData`?

File: client/src/components/screen/ProfileUpdate.tsx [hunk 3]
```diff
@@ -64,6 +66,13 @@ const ProfileImage = styled.Image`
   border-radius: 45px;
 `;
 
+declare let Blob: {
+  prototype: Blob;
+  new (): Blob;
+  new (blobParts: Array<string>, options: Record<string, string>): Blob;
+  name: string;
+};
+
```
Comment (hyochan) [inline client/src/components/screen/ProfileUpdate.tsx:75]: Could you kindly refer to any resource on having this approach? I think It is good but I want to read more about this!

**Review by hyochan — CHANGES_REQUESTED** [review]

File: client/src/components/screen/ProfileUpdate.tsx (hunk 2 above)
Comment (filoscoder) [reply client/src/components/screen/ProfileUpdate.tsx:147]: Not really. I just replace using the plain fetch function here to use the mutation then use the GraphQL fetch.
But I needed to create a `file` to send as a parameter value (`mutationConfig.uploadables.file`) on the mutation, which I choose to do it with the Blob object. (See: https://github.com/jaydenseric/apollo-upload-client/#Blob).
Then this file is appended on the `FormData` from the `GraphQL FetchFunction` (see `fetch.ts`)

File: client/src/components/screen/ProfileUpdate.tsx (hunk 3 above)
Comment (filoscoder) [reply client/src/components/screen/ProfileUpdate.tsx:75]: Sure, first I followed the Blob example to upload a file from the client (See: https://github.com/jaydenseric/apollo-upload-client/#Blob). Then I need to define the Blob type, I get some hints from this to references.
- [Interface File](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.file.html)
- [Blob in TypeScript](https://stackoverflow.com/questions/13124080/blob-in-typescript)

**Review by hyochan — APPROVED** [review]
LGTM!

**Pull request merged**